scalalib/src/mill/javalib/android/AndroidAppModule.scala

```diff
@@ -339,4 +393,63 @@ trait AndroidAppModule extends JavaModule {
 
     PathRef(keystoreFile)
   }
+
+  /**
+   * Runs the Android Lint tool to generate a report on code quality issues.
+   *
+   * This method utilizes Android Lint, a tool provided by the Android SDK,
+   * to analyze the source code for potential bugs, performance issues, and
+   * best practices compliance. It generates a report in the specified format.
+   *
+   * The report is saved in the task's destination directory as "report.html" by
+   * default. It can be changed to other file format such as xml, txt and sarif.
+   *
+   * For more details on the Android Lint tool, refer to:
+   * [[https://developer.android.com/studio/write/lint]]
+   */
+  def androidLintRun() = Task.Command {
+
+    val formats = androidLintReportFmt()
+
+    // Generate the alternating flag and file path strings
+    val reportArg: Seq[String] = formats.toSeq.flatMap { format =>
+      Seq(format.flag, (Task.dest / s"report.${format.extension}").toString)
+    }
+
+    // Set path to generated `.jar` files and/or `.class` files
+    val cp = runClasspath().map(_.path).filter(os.exists).mkString(":")
+
+    // Set path to the location of the project source codes
+    val src = sources().map(_.path / "main" / "java").filter(os.exists).mkString(":")
```

Comment:
I think you shouldn't hardcode path as `main/java`, because both pieces can be actually different. I would suggest to use `Lib.findSourceFiles` (extension can be both `.java` and `.kotlin`) at least with the given `sources`.

Ideally, in order to not have a huge path with each individual files there, it should be a way to find a common roots for the given set of the files, but I don't know if there are utility methods in the project's codebase to compute that.

Reply:
actual files can be under any path of `sources()` root, so hardcoding `main/java` doesn't work: nothing prevents having source files under `sources()/foo/bar/foobar`, for example.